### PR TITLE
chore: add metrics port to mount pod

### PR DESCRIPTION
**Pull request by robgordon89**
Adding metrics port to mount pod, the approach here I'm not entirely happy with the getMetricsPort function, had to do some regex on the options 🤔 not sure how else to tackle it, unless it is moved to its own config.

Added tests to show that it is overridden when you update the metric port in mount options.

Set 9567 as default so it wont break current deployments

### CODE REVIEW

File: pkg/juicefs/mount/builder/pod.go [hunk 1]
```diff
@@ -194,3 +204,20 @@ func (r *Builder) getInitContainer() corev1.Container {
 	})
 	return container
 }
+
+func (r *Builder) getMetricsPort() int32 {
```
Comment (zwwhdls) [inline pkg/juicefs/mount/builder/pod.go:217]: Can you add unit test for this function?

**Review by zwwhdls — CHANGES_REQUESTED** [review]

**Comment by zwwhdls** [comment]
Can you also fix the code scanning alert? Thanks. https://github.com/juicedata/juicefs-csi-driver/security/code-scanning/2

File: pkg/juicefs/mount/builder/pod.go (hunk 1 above)
Comment (robgordon89) [reply pkg/juicefs/mount/builder/pod.go:217]: Sorry for the delay on this 

unit test added

**Comment by zwwhdls** [comment]
Thanks for your contribution! @robgordon89

**Pull request merged**